tools/wave/testing/test_loader.py

```diff
@@ -0,0 +1,192 @@
+from __future__ import absolute_import
+import json
+import os
+import re
+
+AUTOMATIC = u"automatic"
+MANUAL = u"manual"
+
+TEST_TYPES = [AUTOMATIC, MANUAL]
+
+
+class TestLoader(object):
+    def initialize(
+        self,
+        exclude_list_file_path,
+        include_list_file_path,
+        results_manager,
+        api_titles
+    ):
+        self._exclude_list_file_path = exclude_list_file_path
+        self._include_list_file_path = include_list_file_path
+        self._results_manager = results_manager
+        self._tests = {}
+        self._tests[AUTOMATIC] = {}
+        self._tests[MANUAL] = {}
+        self._api_titles = api_titles
+
+    def load_tests(self, manifest_file_path):
```

Comment:
Is this trying to read a `MANIFEST.json` file? You really really want to use the existing libraries for that rather than hand rolling it, because otherwise it will be liable to break whenever the format changes.